### PR TITLE
Feature/aps driver

### PR DESCRIPTION
**Pull request by gribeill**
Auspex driver for the APSI/DACII. Pretty minimal changes but tested on hardware and seems to be working fine. Seems to work fine with both the MinGW and VS2015 libaps -- not yet tested on Linux.

### CODE REVIEW

File: src/auspex/instruments/bbn.py [hunk 1]
```diff
@@ -147,6 +160,231 @@ def __init__(self, name, bases, dct):
...
+    @property
+    def repeat_mode(self):
+        return self._repeat_mode
+    @repeat_mode.setter
+    def repeat_mode(self, mode):
+        mode = mode.upper()
+        if mode not in self._repeat_mode_dict.values():
+            raise ValueError("Unknown repeat mode {} for APS {}. Repeat mode must be one of {}.".format(mode, self.name, list(self._repeat_mode_dict.values())))
+        else:
+            self.wrapper.setRepeatMode(self._repeat_mode_inv_dict[mode])
```
Comment (dieris) [inline src/auspex/instruments/bbn.py]: this requires a channel as input

File: src/auspex/instruments/bbn.py [hunk 2]
```diff
@@ -39,12 +39,12 @@
     aps1 = MagicMock()
 else:
     try:
-        import APS as libaps
```
Comment (dieris) [inline src/auspex/instruments/bbn.py]: @gribeill wasn't this correct?

Comment (dieris) [reply src/auspex/instruments/bbn.py]: nvm, I didn't checkout the latest libaps

File: src/auspex/instruments/bbn.py [hunk 3]
```diff
@@ -33,6 +33,22 @@
...
+    aps1 = MagicMock()
+else:
+    try:
+        import aps as libaps
+        if libaps.APS_PY_WRAPPER_VERSION < 1.4:
+            logger.warning("Old version of libaps found. Using MagicMock instead.")
+            raise ImportError
+        fake_aps1 = False
+    except:
+        fake_aps1 = True
```
Comment (grahamrow) [inline src/auspex/instruments/bbn.py]: Should we add a warning?

File: src/auspex/instruments/bbn.py [hunk 4]
```diff
@@ -195,7 +444,7 @@ def __init__(self, resource_name=None, name="Unlabeled APS2"):
         if fake_aps2:
             self.wrapper = MagicMock()
         else:
-            self.wrapper = aps2.APS2()
+            self.wrapper = APS.APS()
```
Comment (grahamrow) [inline src/auspex/instruments/bbn.py]: was this supposed to happen?

Comment (matthewware) [reply src/auspex/instruments/bbn.py]: Seems like find/replace problem

File: src/auspex/instruments/bbn.py (hunk 3 above)
Comment (matthewware) [reply src/auspex/instruments/bbn.py]: Maybe like we do for the aps2:
https://github.com/BBN-Q/Auspex/blob/feature/aps-driver/src/auspex/instruments/bbn.py#L442

**Comment by matthewware** [comment]
If the libaps isn't loaded on import we get a warning and a warning when the APS get's initialized that we're using MagicMock.  For the APS2 we only get a warning when it get's initialized.  Should we make them consistent?

**Comment by dieris** [comment]
Testing on an APS, I noticed that library calls that do not use channels, like `set_trigger_source` and `set_trigger_interval` are not functional (i.e., do not seem to update the settings). As opposed to `set_run_mode` or `set_repeat_mode`, which for some reason require a library call per channel.

**Comment by gribeill** [comment]
@dieris, I just checked `set_trigger_source` and `set_trigger_interval` on an APS, and the settings are being updated for me? What version of the bitfile is loaded on the APS (you can run `aps.wrapper.get_bitfile_version()` to check)?

**Comment by gribeill** [comment]
PS: you should be using the `py3-compat` branch of libaps... otherwise there are issues with unicode conversion of strings.

**Comment by dieris** [comment]
`aps.wrapper.read_bitfile_version()` returns 511. What do you have? Yes, I'm on the libaps `py3-compat` branch

**Comment by gribeill** [comment]
Huh, I get `4`. Which computer is this on? I'll take a look.

**Comment by dieris** [comment]
As @gribeill found out, solved by manually loading the firmware with a direct library call

**Comment by gribeill** [comment]
Thanks for the input everyone :)

**Pull request merged**